engine/access/rpc/backend/events_index.go

```diff
@@ -40,6 +43,22 @@ func (e *EventsIndex) GetEvents(blockID flow.Identifier, height uint64) ([]flow.
 	return e.events.ByBlockID(blockID)
 }
 
+func (e *EventsIndex) GetEventsByTransactionID(blockID flow.Identifier, height uint64, transactionID flow.Identifier) ([]flow.Event, error) {
```

Comment:
similar comment to https://github.com/onflow/flow-go/pull/5306/#discussion_r1491112450

```suggestion
func (e *EventsIndex) ByBlockIDTransactionID(blockID flow.Identifier, height uint64, transactionID flow.Identifier) ([]flow.Event, error) {
```